src/platform/packages/private/kbn-grid-layout/grid/grid_panel/grid_panel.tsx

```diff
@@ -24,50 +24,18 @@ export interface GridPanelProps {
     panelId: string,
     setDragHandles?: (refs: Array<HTMLElement | null>) => void
   ) => React.ReactNode;
-  interactionStart: (type: PanelInteractionEvent['type'] | 'drop', e: UserInteractionEvent) => void;
   gridLayoutStateManager: GridLayoutStateManager;
 }
 
 export const GridPanel = forwardRef<HTMLDivElement, GridPanelProps>(
-  (
-    { panelId, rowIndex, renderPanelContents, interactionStart, gridLayoutStateManager },
-    panelRef
-  ) => {
+  ({ panelId, rowIndex, renderPanelContents, gridLayoutStateManager }, panelRef) => {
     const [dragHandleApi, setDragHandleApi] = useState<DragHandleApi | null>(null);
     const { euiTheme } = useEuiTheme();
 
-    useEffect(() => {
-      const onDropEventHandler = (dropEvent: MouseEvent) => interactionStart('drop', dropEvent);
-      /**
-       * Subscription to add a singular "drop" event handler whenever an interaction starts -
-       * this is handled in a subscription so that it is not lost when the component gets remounted
-       * (which happens when a panel gets dragged from one grid row to another)
-       */
-      const dropEventSubscription = gridLayoutStateManager.interactionEvent$.subscribe((event) => {
-        if (!event || event.id !== panelId) return;
-
-        /**
-         * By adding the "drop" event listener to the document rather than the drag/resize event handler,
-         * we prevent the element from getting "stuck" in an interaction; however, we only attach this event
-         * listener **when the drag/resize event starts**, and it only executes once (i.e. it removes itself
-         * once it executes, so we don't have to manually remove it outside of the unmount condition)
-         */
-        document.addEventListener('mouseup', onDropEventHandler, {
-          once: true,
-          passive: true,
-        });
-      });
-
-      return () => {
-        dropEventSubscription.unsubscribe();
-        document.removeEventListener('mouseup', onDropEventHandler); // removes the event listener on row change
-      };
-      // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, []);
-
     /** Set initial styles based on state at mount to prevent styles from "blipping" */
     const initialStyles = useMemo(() => {
-      const initialPanel = gridLayoutStateManager.gridLayout$.getValue()[rowIndex].panels[panelId];
+      const initialPanel =
+        gridLayoutStateManager.proposedGridLayout$.getValue()[rowIndex].panels[panelId];
```

Comment:
Should this be `gridLayout$` as well?

Reply:
Yes, it wasn't ready yesterday, I got stuck with some reviews and the performance debugging. But all the responses to your comments have been implemented so it's ready to re-review :)